### PR TITLE
Add optimized IoU function in Cython.

### PR DESCRIPTION
**Pull request by facundo-lezama**
This PR adds a new implementation for the IoU distance calculation. In particular, to speed up the process of calculating distances, we implemented the new module in Cython. This resulted in x10 speed up in some scenarios. 

We now use this optimized IoU compiled module in Norfair. For this purpose, we must ensure the module gets correctly compiled when installing Norfair. For now, we won't be distributing platform-specific wheels since we are not releasing a new version of Norfair. The module will get compiled on the user's end when installing Norfair with pip. In a future PR, we will address the necessary changes to distribute the package with the corresponding precompiled versions for each platform. 

Changes include:
- Implementation of optimized IoU in Cython
- Add build.py for building the new module
  - this gets included in the setup.py when installing Norfair
- Modification of tests accordingly 
- Integration of new IoU distance function with distances API

Pending:
- [X] Rebase when https://github.com/tryolabs/norfair/pull/211 gets merged.
- [x] Integrate Cython IoU with vectorized distances new API
- [x] Check ReID is still working 
- [x] Check linting issue
- [x] Write tests 
- [x] Write PR's description
- [x] Check issue when installing with poetry

### CODE REVIEW

**Review by javiber — COMMENTED** [review]
LGTM will test it on windows just in case

**Review by javiber — APPROVED** [review]
Tested `poetry install` on linux and it works now!
Also tested `pip install git+https://github.com/tryolabs/norfair.git@optimized-iou#egg=norfair` on windows and it worked (after installing the building toolkit for c/c++)

File: build.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from setuptools import Extension
+
+import Cython.Build
+import numpy
+
+
+extension = Extension(
+    "optimized_iou",
```
[the remark this answers is not in the document]
Comment (facundo-lezama) [reply build.py]: I did some changes to keep it under `norfair.optimized_iou`.

**Review by DiegoFernandezC — APPROVED** [review]

**Comment by facundo-lezama** [comment]
Closed this PR since we found out how to implement a vectorized IoU using `numpy`.

**Pull request closed without merging**